test/unit/test_devices/test_aireos_device.py

```diff
@@ -538,6 +640,137 @@ def test_show_list(mock_enable, aireos_send_command_timing):
     device.native.send_command_timing.assert_has_calls([mock.call("send command timing")] * 2)
 
 
+@mock.patch.object(AIREOSDevice, "_ap_images_match_expected")
+@mock.patch.object(AIREOSDevice, "ap_boot_options", new_callable=mock.PropertyMock)
+def test_transfer_image_to_ap_already_active(mock_abo, mock_aime, aireos_device, aireos_boot_image):
+    mock_aime.return_value = True
+    assert aireos_device.transfer_image_to_ap(aireos_boot_image) is False
+    mock_aime.assert_called_once()
+    mock_abo.assert_called_once()
+
+
+@mock.patch.object(AIREOSDevice, "config")
+@mock.patch.object(AIREOSDevice, "_wait_for_ap_image_download")
+@mock.patch.object(AIREOSDevice, "boot_options", new_callable=mock.PropertyMock)
+@mock.patch.object(AIREOSDevice, "_ap_images_match_expected")
+@mock.patch.object(AIREOSDevice, "ap_boot_options", new_callable=mock.PropertyMock)
+def test_transfer_image_to_ap_already_transferred_primary(
+    mock_abo, mock_aime, mock_bo, mock_wait, mock_config, aireos_device, aireos_mock_path, aireos_boot_image
```

Comment:
I suggest a little longer names that correspond to mock_boot_options, mock_image_match, etc.

Reply:
I have updated all of the shortened names